### PR TITLE
added esConsumes to  modules in RecoTracker/FinalTrackSelectors

### PR DESCRIPTION
**Pull request by JamminJones**
#### PR description:

added esConsumes to 8 modules

#### PR validation:

the code compiles

### CODE REVIEW

**Comment by JamminJones** [comment]
the following modules were not modified because of the complexity of the changes needed: AnalyticalTrackSelector, MultiTrackSelector, TrackMVAClassifierBase, DefaultTrackMvaClassifier, and TrackLwtnnCalssifier.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by Dr15Jones** [comment]
This change was made under my supervision.

**Comment by Dr15Jones** [comment]
The failure in the test for RelVal 5.1 is
```
----- Begin Fatal Exception 18-Jul-2020 04:28:57 CEST-----------------------
An exception of category 'NoProxyException' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'reconstruction_befmix_step'
   [2] Prefetching for module TrackCollectionMerger/'generalTracksBeforeMixing'
   [3] Calling method for module TrackListMerger/'jetCoreRegionalStepTracks'
Exception Message:
No data of type "TrackAlgoPriorityOrder" with label "" in record "CkfComponentsRecord"
 Please add an ESSource or ESProducer to your job which can deliver this data.
----- End Fatal Exception -------------------------------------------------
```

File: RecoTracker/FinalTrackSelectors/plugins/TrackListMerger.cc [hunk 1]
```diff
@@ -215,6 +218,7 @@ TrackListMerger::TrackListMerger(edm::ParameterSet const& conf) {
   lostHitPenalty_ = conf.getParameter<double>("LostHitPenalty");
   indivShareFrac_ = conf.getParameter<std::vector<double>>("indivShareFrac");
   std::string qualityStr = conf.getParameter<std::string>("newQuality");
+  priorityToken = esConsumes<TrackAlgoPriorityOrder, CkfComponentsRecord>(edm::ESInputTag("", priorityName_));
```
Comment (Dr15Jones) [inline RecoTracker/FinalTrackSelectors/plugins/TrackListMerger.cc:221]: The line `priorityName_ = conf.getParameter<std::string>("trackAlgoPriorityOrder");` at the end of the constructor needs to be moved to before seeting of `priorityToken`

**Comment by Dr15Jones** [comment]
please test

**Comment by perrotta** [comment]
Curiously, with this PR the Warning LogMessage about different template ID between 1D and 2D from PixelCPEClusterRepair is issued by a different module than in the baseline. E.g.:

- In PR by PostModuleEvent:
```
%MSG-w PixelCPEClusterRepair:  PostModuleEvent 18-Jul-2020 18:48:26 CEST  Run: 1 Event: 3
different template ID between 1D and 2D 2309 2300
%MSG
```
- In baseline by TrackProducer:initialStepTracksPreSplitting
```
%MSG-w PixelCPEClusterRepair:  TrackProducer:initialStepTracksPreSplitting  18-Jul-2020 14:58:33 CEST Run: 1 Event: 3
different template ID between 1D and 2D 2309 2300
%MSG
```

Out of curiosity, @Dr15Jones do you have an idea why?

**Comment by Dr15Jones** [comment]
> Curiously, with this PR the Warning LogMessage about different template ID between 1D and 2D from PixelCPEClusterRepair is issued by a different module than in the baseline.

My guess is the prefetching might be triggering an ESProducer which normally might not always be run to get some data type (but is always gotten for another one) and therefore the module is running in a different order.

The EventSetup modules can use MessageLogger but MessageLogger is never informed when a ES module is run (it only knows about ED modules).

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass
- Differences in DQM only related to different orders of MessageLogger messages

**Comment by silviodonato** [comment]
+1

**Pull request merged**